mlir-tensorrt/tensorrt/test/Target/TensorRT/linspace.mlir

```diff
@@ -27,3 +24,14 @@ func.func @trt_fill_linspace_dynamic() -> tensor<1024x1024xf32> {
   %0 = tensorrt.linspace [%start:tensor<f32>][%shape:tensor<2xi32>][%step:tensor<2xf32>] : tensor<1024x1024xf32>
   return %0 : tensor<1024x1024xf32>
 }
+
+func.func @trt_fill_linspace_dynamic_dim(%arg0: tensor<?x32xf32> {tensorrt.shape_profile = #tensorrt.shape_profile<min = [1, 32], opt = [64, 32], max = [64, 32]>}) -> tensor<?xf32> {
+    %cst_i32 = tensorrt.constant dense<0> : tensor<1xi32>
+    %0 = tensorrt.shape %arg0 : tensor<?x32xf32> -> tensor<2xi32>
+    %1 = tensorrt.gather {axis = 0 : i64} ins(%0, %cst_i32 : tensor<2xi32>, tensor<1xi32>) -> tensor<1xi32>
+    %2 = tensorrt.linspace[ 0.000000e+00] [%1 : tensor<1xi32>] [ 1.000000e+00] : tensor<?xi32>
+    %3 = tensorrt.identity %2 : tensor<?xi32> to tensor<?xf32>
```

Comment:
Why not use `tensorrt.cast` which is preferred for casting.

Reply:
I took this test case as it is from filed issue.